scripts/check-backwards-compatability.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
```

Comment:
I wish I had a guide for best practices with shell scripts but anyway, here are the usual things that I find important:

1. Use `/usr/bin/env bash` because it allows calling e.g. the brew-installed version of bash rather than the antique version 3.x provided by Apple.
2. Start with `set -eu -o pipefail` (or just `set -eu` if you don't use pipes). These are all great defaults.
3. Name the file without a `.sh` suffix - it's a command for a user who doesn't particularly care in which language it's written.
4. Quote all variable expansions unless you want to split them: `"$foo"`, not `$foo` except in cases like `for x in $items; do ...`. Same for command substitutions: use `echo "$(cat file)"`, not `echo $(cat file)` unless you want to squash the whitespace.

I forgot what the issues are with arrays in bash 3.x but my conclusion was that it made them mostly unusable. Maybe it's fine for what you're doing below.

General practices:
* start the file with a comment that explains what it's purpose is in the universe

Reply:
thanks!

1. good call, fixed
2. added `set -u` and a comment about why I'm not using `-eo pipefail` (I want to run all tags rather than error out on the first thing.
3. I have this as a habit because of the various tools (like semgrep) that use extension to make decision about static analysis
4. agreed, fixed
5. added a comment